lib/snapshot/BUILD.gn

```diff
@@ -280,9 +281,19 @@ if (host_os == "mac" && (target_cpu == "arm" || target_cpu == "arm64")) {
 }
 
 if (host_os == "mac" && target_os == "mac") {
-  macos_gen_snapshots("create_macos_gen_snapshots") {
-    deps = [ ":generate_snapshot_bin" ]
-    arch = "$target_cpu"
+  strip_bitcode("create_macos_gen_snapshots") {
+    # The toolchain-specific output directory. For cross-compiles, this is a
+    # clang-x64 or clang-arm64 subdirectory of the top-level build directory.
+    host_output_dir = get_label_info(
+            "//third_party/dart/runtime/bin:gen_snapshot($host_toolchain)",
+            "root_out_dir")
+
+    input = "${host_output_dir}/gen_snapshot"
+    output = "${root_out_dir}/gen_snapshot_${target_cpu}"
```

Comment:
This is the same output path that we've been historically emitted this binary. In a future patch, I'll make this consistent with the `create_arm_gen_snapshot` target, which emits it alongside the original in `host_output_dir`. 

Not doing it in this patch since I'll need to verify expected paths used by the recipes first, and migrate stepwise if needed.